Sockets/Sockets.Implementation.NET/TcpSocketClient.cs

```diff
@@ -36,11 +38,41 @@ internal TcpSocketClient(TcpClient backingClient)
         /// </summary>
         /// <param name="address">The address of the endpoint to connect to.</param>
         /// <param name="port">The port of the endpoint to connect to.</param>
-        public async Task ConnectAsync(string address, int port)
+        /// <param name="secure">Is this connection secure?</param>
+        public async Task ConnectAsync(string address, int port, bool secure = false)
         {
             await _backingTcpClient.ConnectAsync(address, port);
+            if (!secure)
+            {
+                _secureStream = new SslStream(_backingTcpClient.GetStream(), true, (sender, cert, chain, sslPolicy) => ServerValidationCallback(sender,cert,chain,sslPolicy));
+                _secureStream.AuthenticateAsClient(address, null, System.Security.Authentication.SslProtocols.Tls, false);
```

Comment:
If `AuthenticateAsClient` throws an exception (and the user catches), we will end up with an invalid `_secureStream`. Can we use a local variable until after `AuthenticateAsClient` is successful,  and then assign to `_secureStream`?